### ISSUE TITLE
Move the plugin to Roundcube core (replace markasjunk)

### ISSUE THREAD

**Issue by alecpl** [johndoh/roundcube-markasjunk2#59]
I've been asked if core's markasjunk plugin could be easily replaced with your plugin. So, I have questions to you:
1. What do you think about this?
2. Can your plugin (with default configuration) work as markasjunk plugin does? I don't want to make the switch a big regression or too much trouble for current markasjunk plugin users.

Would be good to know the answers before I put my hands on this.

**Comment by johndoh** [comment]
Out of the box they are reasonably similar. I did some tests, in markasjunk when the Junk action is executed then a flag of `Junk` is set and a flag of `NonJunk` is removed. In markasjunk2 a flag of `Junk` is set but that’s it. The default could easily be changed to also include the `NonJunk` flag though.

I also noticed a few flaws in the current way flags work in markasjunk2 right now so I’ll work on some improvements there and probably bring it in line with markasjunk, as regardless of if its promoted to the core I think they should match.

Other than that the only other real difference is the addition of the “not junk” button. An option could be added to remove the “not junk” button. I do not think it should be hidden by default though as current users of markasjunk2 are used to that button.

There is one trivial difference in the UI. In en_GB markasjunk2 uses “Spam” (same as in the core) while markasjunk uses “Junk”.

If markasjunk2 moved from being a third-party plugin to a core one then there are 1 or 2 bits of integration with other plugins which should be moved out of markasjunk2 and into the respective plugins to match how integration works with other core plugins.
